packages/opds-web-client/src/components/Book.tsx

```diff
@@ -11,8 +11,6 @@ export interface BookProps {
   book: BookData;
   collectionUrl?: string;
   updateBook: (url: string) => Promise<BookData>;
-  fulfillBook: (url: string) => Promise<Blob>;
-  indirectFulfillBook: (url: string, type: string) => Promise<string>;
```

Comment:
@adriana-alter , as part of the redux and hook update, we don't need to pass down these actions as props down from the root to the leaf node. In this specific case, these two functions were being passed from 
```
Root.tsx -> BookDetails (which inherits from Book) -> DownloadButton.tsx
```

The two functions as props were not being as at all in BookDetails or Book so now that's why this is being removed here. Now, only DownloadButton knows about these actions and calls them. The code reduction is nice, but I do see how this makes DownloadButton less reusable _in other apps_, but works as intended in for this app.

Reply:
I've been thinking about this too, and it's true that `useSelector` couples a component to the redux implementation within the app, while `connect` is creating a wrapper component that deals with the logic outside of the presentation component, which in this case would be the button. 

I think there are valid use cases for both `useSelector` and `connect`. In this case, if OPDS was meant to be a UI component library, it would make sense to have a series of UI-only components like `DownloadButton`. But in my experience, it's the logic in OPDS that is useful, not usually the UI components. I think that's the same for simply-e and circulation-web. So the goal shouldn't be to make all the UI components reusable, but the logic reusable:
```js
const MyDownloadButton = ({book}) => {
    // gets fulfill, dispatch, etc internally
    const {label, onClick, ...etc} = useDownloadButton(book);
    const myOnClick = () => {
        // something custom
        onClick();
    }
    return (
        <CustomButtonUI onClick={myOnClick} {...etc}>{label}</CustomButtonUI>
    );
}
```
So yes, `connect` is better at decoupling a component from redux, but at the cost of some boilerplate and indirection. We should consider what decoupling is actually useful for consumers of OPDS, and in my experience it isn't making it so the UI components can be used without a redux. The first step in making any of the logic reusable is getting rid of that `mergeRootProps` file, so I think this is definitely a step in the right direction.

Just my 2 cents after working in CPW 🙂

Reply:
Yes, that's nicely worded. Those buttons aren't necessarily used in the admin as much as the logic and type interfaces, so it shouldn't be a big deal regarding coupling and reusability. I also wanted to make sure I was able to describe the trade-off that is happening due to this refactoring, so thanks for the verification and longer explanation.